layers.go

```diff
@@ -970,6 +985,87 @@ func (r *layerStore) SetNames(id string, names []string) error {
 	return ErrLayerUnknown
 }
 
+func (r *layerStore) datadir(id string) string {
+	return filepath.Join(r.layerdir, id)
+}
+
+func (r *layerStore) datapath(id, key string) string {
+	return filepath.Join(r.datadir(id), makeBigDataBaseName(key))
+}
+
+func (r *layerStore) BigData(id, key string) ([]byte, error) {
```

Comment:
How big is “big”? Does it make sense for the API to read/write using `[]byte`, or should that be a streaming interface?

Reply:
For a bit more context, this is probably not a concern for fairly fixed things like manifests (in image “big data”). OTOH there _has_ been a report of Raspberry Pis running out of memory, probably when loading tar-split metadata into RAM.

Reply:
"big" is not really that big, the generated manifest file for Fedora is around 3M

Reply:
We’ve had reports like https://github.com/containers/image/issues/495 (“hundreds of gigabytes”); and on the other side, users on Raspberry Pi with IIRC 256 MB of RAM.

I’m not yet saying that this is not good enough, right now I’m just pointing out that Fedora is _by far_ not the worst case of how far this should scale.

Reply:
I echo @mtrmac's concern.  Returning an `io.ReadCloser` seems like a safe(r) bet.